### PR TITLE
redesign popup assignee

### PR DESCRIPTION
**Pull request by pallabez**
### Dev Tested:

- [x] Yes
- [ ] No

### Put the Feature behind Feature Flag

- [ ] Yes
- [x] No

#### After 
image
image

### Follow-up Issues (if any):

> NOTE: After your PR is merged in main please verify its working on prod and raise a dev to main PR

### CODE REVIEW

File: adapters/user.transformer.ts [hunk 1]
```diff
@@ -8,13 +8,19 @@ export const transformUsers = (users: any): User[] => {
...
     lastName: user.last_name,
-    githubDisplayName: user.github_display_name,
-    githubId: user.github_id,
-    avatarUrl: user.picture?.url
+    avatar: user.picture?.url,
+    initials: displayName.trim()
+      .toUpperCase()
+      .split(" ", 2)
+      .map((str) => str.charAt(0))
+      .join(""),
```
Comment (bhtibrewal) [inline adapters/user.transformer.ts:24]: if we wanted initials only why took this long approach 
```
initials: 
`${user.first_name.chatAt(0) user.last_name.charAt(0)}`.toUpperCase()
```

File: components/PopupAssignee.vue [hunk 2]
```diff
@@ -1,50 +1,91 @@
...
+const state = reactive({
+  search: "",
+  isMenuOpen: false
 })
+const filteredUsers = computed(() => fuzzySearch(users, state.search))
+
+function fuzzySearch(users: User[] , searchText: string) {
+  if (!searchText || searchText == assignee.value?.displayName) {
+    return users;
+  }
```
Comment (bhtibrewal) [inline components/PopupAssignee.vue:78]: didn't get how the filtering and mapping is happening here

File: adapters/goal.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ const goalSiteConfig = {
 
 export const fetchGoals = async (): Promise<Goal[]> => {
   const goals: Goal[] = await axios
-    .get(`${API.GOAL_BASE_URL}/goal/`)
+    .get(`${API.GOAL_BASE_URL}/api/v1/goal/`)
```
Comment (bhtibrewal) [inline adapters/goal.ts:15]: let's include `/api/v1/` in GOAL_BASE_URL

File: adapters/user.transformer.ts (hunk 1 above)
Comment (pallabez) [reply adapters/user.transformer.ts:24]: Make sense. Thanks for the suggestion.

File: components/PopupAssignee.vue (hunk 2 above)
Comment (pallabez) [reply components/PopupAssignee.vue:78]: Will make some changes and add a comment here.

File: adapters/goal.ts (hunk 3 above)
Comment (pallabez) [reply adapters/goal.ts:15]: I think `/api` in the GOAL_BASE_URL will do as v1 can change to v2 someday

**Review by bhtibrewal — APPROVED** [review]

**Pull request merged**